samza-core/src/main/java/org/apache/samza/zk/ZkKeyBuilder.java

```diff
@@ -51,6 +51,7 @@
 
   static final String PROCESSORS_PATH = "processors";
   static final String JOBMODEL_GENERATION_PATH = "JobModelGeneration";
+  static final String JOB_MODEL_UPGRADE_BARRIER = "jobModelUpgradeBarrier";
 
   public ZkKeyBuilder(String pathPrefix) {
     if (Strings.isNullOrEmpty(pathPrefix)) {
```

Comment:
Thanks for pointing out. I also found some inaccurate unit tests for this case and another bug in Line 91. Let me create a separate PR for this.